packages/opentelemetry-core/src/baggage/propagation/W3CBaggagePropagator.ts

```diff
@@ -58,7 +58,8 @@ export class W3CBaggagePropagator implements TextMapPropagator {
   }
 
   extract(context: Context, carrier: unknown, getter: TextMapGetter): Context {
-    const headerValue: string = getter.get(carrier, BAGGAGE_HEADER) as string;
+    const headerBaggage = getter.get(carrier, BAGGAGE_HEADER);
+    const headerValue = Array.isArray(headerBaggage) ? headerBaggage.join(',') : headerBaggage;
```

Comment:
I think `headerValue` and `baggageString` might be more descriptive names for these variables.